src/subscription/jobs/rates-notification.job.ts

```diff
@@ -1,40 +1,48 @@
 import { config } from '../../config';
 import { SubscriptionsRepository } from '../data-access/repositories/subscription.repository';
-import { EmailService } from '../../common/services/email.service';
 import { ExchangerService } from '../../rate/service/exchanger.service';
 import { MonobankClient } from '../../rate/data-access/exchangers/monobank-client';
 import { NBUClient } from '../../rate/data-access/exchangers/nbu-client';
 import { Privat24Client } from '../../rate/data-access/exchangers/privat24-client';
 import logger from '../../common/services/logger.service';
 import { SubscriptionsService } from '../service/services/subscription.service';
 import { BanksExchangeHandler } from '../../rate/service/bank-exchange-handler';
+import { setupEventProducer } from '../../common/services/messaging/event-producer';
+import { v4 as uuidv4 } from 'uuid';
+import { SystemEventType } from '../../common/models/system-event.model';
 
 export async function sendDailyRateEmail() {
   try {
     const subscriptionService = new SubscriptionsService(new SubscriptionsRepository());
-    const emailService = new EmailService();
     const monobankHandler = new BanksExchangeHandler(new MonobankClient());
     const nbuHandler = new BanksExchangeHandler(new NBUClient());
     const privat24Handler = new BanksExchangeHandler(new Privat24Client());
     const bankExchangeHandler = monobankHandler.setNext(nbuHandler).setNext(privat24Handler);
     const requestsLimit = config.api.emailServer.rateLimit;
 
+    const messageProducer = await setupEventProducer();
+
     logger.info(`[Scheduled job] [${new Date()}] going to send emails to subsribed users`);
     const currentRate = await new ExchangerService(bankExchangeHandler).getCurrentRate();
     let subcriptions = await subscriptionService.getAll({ limit: requestsLimit });
     let hasMore = !!subcriptions.length;
     const failedToSendEmailErrors: string[] = [];
     while (hasMore) {
       const sendEmailPromises = subcriptions.map((subscription) => {
-        return emailService.sendCurrencyRateEmail({ to: subscription.email, currencyRate: currentRate });
+        return messageProducer.sendEvent(config.messageBroker.topics.email, {
```

Comment:
I think that it is worth somehow simplifying the sending of an event here because { eventId: string, eventType: string, timestamp: Date, data: any } is already a contract by which all events will be sent to the queue, in which only eventType and data will differ